File: code-examples/c/authentication/sha1.c
#include <bson/bson.h>
#include <mongoc/mongoc.h>
#include <stdio.h>

// Replace the following with values for your environment.
char *username = "<username>";
char *password = "<password>";
char *clusterUrl = "<MongoDB cluster url>";

char *uri;

// SCRAM-SHA-1 authentication mechanism
char *authMechanism = "SCRAM-SHA-1";

int main (int argc, char *argv[]) {
    mongoc_client_t *client;
    mongoc_database_t *database;
    bson_error_t error;

    mongoc_init();

    // Create connection URL
    char *encodedUsername = mongoc_uri_encode (username);
    char *encodedPassword = mongoc_uri_encode (password);
    asprintf(&uri, "mongodb+srv://%s:%s@%s/?authMechanism=%s", 
                 encodedUsername, encodedPassword, clusterUrl, authMechanism);

    // Create a new MongoClient
    client = mongoc_client_new (uri);

    // Connecting to the server
    if (!mongoc_client_get_server_status (client, NULL, NULL, &error)) {
        fprintf (stderr, "Error with connection: %s\n", error.message);
        return EXIT_FAILURE;
    } else {
        printf ("Connected successfully to server\n");
    }

    // Ensures that the client will close when you finish/error
    mongoc_client_destroy (client);
    mongoc_cleanup();

    free(encodedUsername);
    free(encodedPassword);
    free(uri);

    return EXIT_SUCCESS;
}

File: code-examples/php/authentication/sha1.php
<?php

require 'vendor/autoload.php';

// Replace the following with values for your environment.
$username = "<username>";
$password = "<password>";
$clusterUrl = "<MongoDB cluster url>";

$authMechanism = "SCRAM-SHA-1";

// Replace the following with your MongoDB deployment's connection string.
$uri = sprintf('mongodb://%s:%s@%s/?authMechanism=%s', 
    $username, 
    $password, 
    $clusterUrl, 
    $authMechanism
);

// Create a new MongoDB driver manager instance.
$manager = new MongoDB\Driver\Manager($uri);

function run($manager) {
    try {
        // Establish and verify connection
        $command = new MongoDB\Driver\Command(['ping' => 1]);
        $manager->executeCommand("admin", $command);

        echo "Connected successfully to server";
    } catch (MongoDB\Driver\Exception\Exception $e) {
        echo $e->getMessage(), "\n";
        echo $e->getTraceAsString(), "\n";
    } finally {
        // There's no need to "close" the connection in MongoDB PHP driver
    }
}

run($manager);
?>
